### PR TITLE
[API Deprecate]Rename DGLHeterpGraph to DGLGraph in python files

### PR DESCRIPTION
**Pull request by peizhou001**
## Description

This PR change ```DGLHeterpGraph``` to ```DGLGraph``` in Python files, changes in docs are in a separate PR.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/compute/test_batched_heterograph.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def test_topology(gs, idtype):
 
 @parametrize_idtype
 def test_batching_batched(idtype):
-    """Test batching a DGLHeteroGraph and a BatchedDGLHeteroGraph."""
+    """Test batching a DGLGraph and a BatchedDGLHeteroGraph."""
```
Comment (mufeili) [inline tests/compute/test_batched_heterograph.py:115]:
```suggestion
    """Test batching a DGLGraph and a batched DGLGraph."""
```

File: python/dgl/__init__.py [hunk 2]
```diff
@@ -36,8 +36,8 @@
 from .batch import *
 from .convert import *
 from .generators import *
-from .heterograph import DGLHeteroGraph
-from .heterograph import DGLHeteroGraph as DGLGraph  # pylint: disable=reimported
+from .heterograph import DGLGraph
+from .heterograph import DGLGraph as DGLHeteroGraph  # pylint: disable=reimported
```
Comment (mufeili) [inline python/dgl/__init__.py:40]: Do we still need to import `DGLHeteroGraph`?

File: python/dgl/batch.py [hunk 3]
```diff
@@ -4,12 +4,12 @@
 from . import backend as F
 from .base import ALL, is_all, DGLError, dgl_warning, NID, EID
 from .heterograph_index import disjoint_union, slice_gidx
-from .heterograph import DGLHeteroGraph
+from .heterograph import DGLGraph
 from . import convert
 from . import utils
 
 
-__all__ = ['batch', 'unbatch', 'slice_batch', 'batch_hetero', 'unbatch_hetero']
```
Comment (mufeili) [inline python/dgl/batch.py:12]: It seems that you also deprecate `batch_hetero` and `unbatch_hetero` in this PR. Make sure you state that in the PR description and link appropriate tracker items.

File: python/dgl/propagate.py [hunk 4]
```diff
@@ -3,7 +3,6 @@
 
 from . import backend as F
 from . import traversal as trv
-from .heterograph import DGLHeteroGraph
```
Comment (mufeili) [inline python/dgl/propagate.py:6]: Is this not used at all?

**Comment by frozenbugs** [comment]
grep DGLHeteroGraph `find . -name *.py | grep -v ./third_party/` | cut -d ':' -f 1 | uniq

Says there are 44 files with DGLHeteroGraph, just to reminder you double check.

File: python/dgl/transforms/functional.py [hunk 5]
```diff
@@ -68,13 +68,11 @@
     'to_block',
     'to_simple',
     'to_simple_graph',
-    'as_immutable_graph',
     'sort_csr_by_tag',
     'sort_csc_by_tag',
     'metis_partition_assignment',
     'partition_graph_with_halo',
     'metis_partition',
-    'as_heterograph',
```
Comment (mufeili) [inline python/dgl/transforms/functional.py:77]: Again, describe these deprecated APIs in the PR description and link appropriate tracker items.

File: python/dgl/traversal.py [hunk 6]
```diff
@@ -4,7 +4,6 @@
 from . import backend as F
 from . import utils
 from ._ffi.function import _init_api
-from .heterograph import DGLHeteroGraph
```
Comment (mufeili) [inline python/dgl/traversal.py:7]: Is this not used at all?

**Comment by mufeili** [comment]
See if you want to have a look @jermainewang

File: tests/compute/test_batched_heterograph.py (hunk 1 above)
Comment (peizhou001) [reply tests/compute/test_batched_heterograph.py:115]: Nice catch, fixed

File: python/dgl/batch.py (hunk 3 above)
Comment (peizhou001) [reply python/dgl/batch.py:12]: Revert it in this PR as it beyond scope.

File: python/dgl/transforms/functional.py (hunk 5 above)
Comment (peizhou001) [reply python/dgl/transforms/functional.py:77]: revert

File: python/dgl/propagate.py (hunk 4 above)
Comment (peizhou001) [reply python/dgl/propagate.py:6]: Readded for assertion.

File: python/dgl/traversal.py (hunk 6 above)
Comment (peizhou001) [reply python/dgl/traversal.py:7]: Readded for assertion.

**Review by mufeili — APPROVED** [review]
I'm good

**Review by frozenbugs — APPROVED** [review]

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**